crates/uv-resolver/src/lock/target.rs

```diff
@@ -155,12 +157,57 @@ impl<'env> InstallTarget<'env> {
         build_options: &BuildOptions,
         install_options: &InstallOptions,
     ) -> Result<Resolution, LockError> {
+        /// Convert a lockfile entry to an installable distribution.
```

Comment:
I'm pretty unhappy with basically this whole method. It's just a bit tortured to achieve the representations we want (e.g., we have to retain nodes even though we won't install them).

Reply:
What about replacing the macro with a method? https://gist.github.com/konstin/da92fa38a8840050b8b2dedfefaef559

```rust
    /// Convert a lockfile entry to an installable distribution.
    fn package_to_node(
        &self,
        package: &Package,
        tags: &Tags,
        build_options: &BuildOptions,
        install_options: &InstallOptions,
    ) -> Result<Node, LockError> {
        if install_options.include_package(
            package.name(),
            self.project_name(),
            self.lock().members(),
        ) {
            let dist = package.to_dist(
                self.workspace().install_path(),
                TagPolicy::Required(tags),
                build_options,
            )?;
            let version = package.version().clone();
            let dist = ResolvedDist::Installable { dist, version };
            let hashes = package.hashes();
            Ok(Node::Dist {
                dist,
                hashes,
                install: true,
            })
        } else {
            let dist = package.to_dist(
                self.workspace().install_path(),
                TagPolicy::Preferred(tags),
                &BuildOptions::default(),
            )?;
            let version = package.version().clone();
            let dist = ResolvedDist::Installable { dist, version };
            let hashes = package.hashes();
            Ok(Node::Dist {
                dist,
                hashes,
                install: false,
            })
        }
    }
```

Reply:
tbh i kinda like having the full graph, it makes reasoning easier.

Another option here would be to have the `install` outline as a mask, e.g. a hashset of node indexes that represent the installed subset; haven't checked if that actually helps though.

Reply:
Good call.